packet/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -282,6 +283,21 @@ storage:
           kind: KubeletConfiguration
           cgroupDriver: "$${docker_cgroup_driver}"
           EOF
+    - path: /opt/etcd-private-interface
+      filesystem: root
+      mode: 0744
+      contents:
+        inline: |
+          #!/bin/bash
+          set -e
+          if ! test -f /tmp/etcd-private-interface; then
+            exit 0
+          fi
+          ipaddress=$(curl --retry-delay 1 --retry 60 --retry-connrefused --retry-max-time 60 --connect-timeout 20 metadata.packet.net/metadata | jq -r ".network.addresses[] | select(.public == false) | .address")
+          sed -i "s|REPLACE_WITH_PRIVATE_IP|$${ipaddress}|g" /etc/systemd/system/etcd-member.service.d/40-etcd-cluster.conf
+          systemctl daemon-reload
+          touch /tmp/etcd-private-interface
```

Comment:
Ah, I forgot, maybe it's better to not rely on `exit 1` to restart it after some time but rather directly restart the service:
```
touch /tmp/etcd-private-interface
systemctl daemon-reload
systemctl restart etcd-member &
exit 1
```